omni_wheel_controller:
  ros__parameters:
    # Joints in your URDF
    omni_wheel_names: ['wheel0_shaft_joint', 'wheel1_shaft_joint', 'wheel2_shaft_joint', 'wheel3_shaft_joint']

    # If you're using an X-configuration, angles are typically ±45°, ±135° (in radians):
    omni_wheel_angle: [0.785398, 2.356194, -2.356194, -0.785398]

    # Distance from center to each wheel [m]
    omni_wheel_distance: 0.3347  # 33.47 cm

    # Wheel radius [m]
    wheel_radius: 0.0762         # 3 inches = 7.62 cm

    # Frame settings
    odom_frame_id: odom
    base_frame_id: base_link
    enable_odom_tf: true

    # Covariances (example values)
    pose_covariance_diagonal:  [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
    twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

    # Odom publish rate
    publish_rate: 50.0

    # Timeout and velocity-limiting
    cmd_vel_timeout: 3.0
    publish_limited_velocity: true
    velocity_rolling_window_size: 10
    publish_cmd: false

    # Velocity/acceleration limits
    linear:
      x:
        has_velocity_limits: true
        max_velocity: 1.0   
        min_velocity: -1.0  
        has_acceleration_limits: false
        has_jerk_limits: false

    angular:
      z:
        has_velocity_limits: true
        max_velocity: 2.0
        min_velocity: -2.0
        has_acceleration_limits: false
        has_jerk_limits: false
